app/Database/CourseInsertion.hs

```diff
@@ -39,10 +43,28 @@ saveGraphJSON jsonStr nameStr = do
                 insertMany_ $ map (\path -> path {pathGraph = gId}) paths
             return $ toResponse $ ("Success" :: String)
 
--- | Inserts course into the Courses table.
+
+-- | Check for duplicate key, then if unique, insert course into Courses table
+--safeInsertCourse :: Course -> ReaderT SqlBackend m (Maybe Course) -- this screws up 'insert'
+--safeInsertCourse :: Course -> ReaderT SqlBackend m () -- this screws up 'insert'
```

Comment:
If this program succesfully compiles, but you don't know what the type is, then you should load this module into the REPL, and then inspect the type directly with `:type safeInsertCourse`. It'll tell you what type was inferred, which is almost always the correct type that you want to annotate the function with.